tests/scripts/generate_psa_tests.py

```diff
@@ -393,46 +425,167 @@ def keys_for_algorithm(self, alg: str) -> Iterator[StorageKey]:
         descr = re.sub(r'PSA_ALG_', r'', alg)
         descr = re.sub(r',', r', ', re.sub(r' +', r'', descr))
         usage = 'PSA_KEY_USAGE_EXPORT'
-        key1 = StorageKey(version=self.version,
-                          id=1, lifetime=0x00000001,
-                          type='PSA_KEY_TYPE_RAW_DATA', bits=8,
-                          usage=usage, alg=alg, alg2=0,
-                          material=b'K',
-                          description='alg: ' + descr)
-        yield key1
-        key2 = StorageKey(version=self.version,
-                          id=1, lifetime=0x00000001,
-                          type='PSA_KEY_TYPE_RAW_DATA', bits=8,
-                          usage=usage, alg=0, alg2=alg,
-                          material=b'L',
-                          description='alg2: ' + descr)
-        yield key2
+        yield StorageKey(version=self.version,
+                         id=1, lifetime=0x00000001,
+                         type='PSA_KEY_TYPE_RAW_DATA', bits=8,
+                         usage=usage, alg=alg, alg2=0,
+                         material=b'K',
+                         description='alg: ' + descr)
+        yield StorageKey(version=self.version,
+                         id=1, lifetime=0x00000001,
+                         type='PSA_KEY_TYPE_RAW_DATA', bits=8,
+                         usage=usage, alg=0, alg2=alg,
+                         material=b'L',
+                         description='alg2: ' + descr)
 
     def all_keys_for_algorithms(self) -> Iterator[StorageKey]:
         """Generate test keys covering algorithm encodings."""
         algorithms = sorted(self.constructors.algorithms)
         for alg in self.constructors.generate_expressions(algorithms):
             yield from self.keys_for_algorithm(alg)
 
+    def generate_all_keys(self) -> List[StorageKey]:
```

Comment:
Should return an iterator like the others.

Reply:
I tried using iterators but it makes the script significantly slower to run, about nine times. Even the use of `itertools.chain()` is not reducing the runtime. I am not investigated what causing this slowdown because the lack of time.
Should I do the change even the cost of the runtime?

Reply:
The script is getting quite complex and doesn't run often, so I'd rather have it easy to read than fast. If it gets annoyingly slow, we can do a pass of optimization later.